### PR TITLE
Added unit tests and fixed a couple of bugs.

### PR DESCRIPTION
**Pull request by cableramki**

### CODE REVIEW

File: src/json_interface.c [hunk 1]
```diff
@@ -31,17 +31,12 @@
 /*----------------------------------------------------------------------------*/
 /*                               Data Structures                              */
 /*----------------------------------------------------------------------------*/
-typedef struct __ll_node {
-    char *entry;
-    char *value;
-    struct __ll_node *next;
-} ll_t;
+ll_t *head = NULL;
```
Comment (cableramki) [inline src/json_interface.c]: Removed static to allow for unit testing.

File: src/json_interface.h [hunk 2]
```diff
@@ -24,7 +24,11 @@ extern "C" {
 /*----------------------------------------------------------------------------*/
 /*                               Data Structures                              */
 /*----------------------------------------------------------------------------*/
-/* none */
+typedef struct ll_node {
+    char *entry;
+    char *value;
+    struct ll_node *next;
+} ll_t;
```
Comment (cableramki) [inline src/json_interface.h]: Moved to header to allow for unit testing.

File: src/json_interface.h [hunk 3]
```diff
@@ -24,7 +24,11 @@ extern "C" {
...
```
Comment (cableramki) [inline src/json_interface.h]: Moved to header to allow for unit testing.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -60,6 +49,17 @@ target_link_libraries (test_wrp_interface_free -lpthread -lwrp-c -lmsgpackc -ltr
...
+#   test_json_interface
+#-------------------------------------------------------------------------------
+add_test(NAME test_json_interface COMMAND test_json_interface)
+add_executable(test_json_interface test_json_interface.c ../src/json_interface.c)
+set_property(TARGET test_json_interface PROPERTY C_STANDARD 99)
+
+target_link_libraries (test_json_interface -lpthread -lmsgpackc -ltrower-base64 -lcjson -lcimplog -lwrp-c -lnanomsg)
+target_link_libraries (test_json_interface gcov)
+target_link_libraries (test_json_interface cunit)
+
```
Comment (schmidtw) [inline tests/CMakeLists.txt]: Did any of this change or just move?

Comment (cableramki) [inline tests/CMakeLists.txt]: Just moved.

**Comment by cableramki** [comment]
Closing to resubmit.

**Pull request closed without merging**